src/mdns/mdns_mdnssd.cpp

```diff
@@ -338,7 +338,16 @@ void PublisherMDnsSd::Process(const MainloopContext &aMainloop)
         {
             otbrLogWarning("DNSServiceProcessResult failed: %s (serviceRef = %p)", DNSErrorToString(error), serviceRef);
         }
+        if (error == kDNSServiceErr_ServiceNotRunning)
+        {
+            otbrLogWarning("Need to reconnect to mdnsd");
+            Stop();
```

Comment:
Is `DNSServiceRef` in `readyServices` still valid  after Stop?

Reply:
No.

Reply:
If so, should we quit this loop?

Reply:
I've already made a `ExitNow()` in this if statement. I guess it's sufficient?